### PR TITLE
Fix remaining custom format issues

### PR DESCRIPTION
**Pull request by sirDonovan**
Rules for custom formats are once again sanitized and can be passed around more easily by attaching them to the format's id.

Fixes:
- "Rule:" being required to add or remove a rule
- ruleset changes not being shown in battles
- the display of custom rules being affected by user input
- custom formats being broken by /hotpatch formats

### CODE REVIEW

**Comment by Zarel** [comment]
Oh hey, you used my idea about `@@@`.

I don't know how to feel about it, but it certainly makes the code easier.

**Comment by MacChaeger** [comment]
Oh this would make my pr unnecessary wouldn't it?

**Comment by Zarel** [comment]
Possibly.

File: team-validator.js [hunk 1]
```diff
@@ -1162,32 +1162,28 @@ class TeamValidatorManager extends ProcessManager {
...
 		let removeNicknames = message.substr(pipeIndex + 1, nextPipeIndex - pipeIndex - 1);
 		let team = message.substr(nextPipeIndex + 1);
 
-		process.send(id + '|' + this.receive(format, customRules, removeNicknames, team));
+		process.send(id + '|' + this.receive(format, removeNicknames, team));
 	}
 
-	receive(format, customRules, removeNicknames, team) {
+	receive(format, removeNicknames, team) {
+		let customRules = format.includes('@@@') ? format.split('@@@')[1].split(',') : null;
```
Comment (Zarel) [inline team-validator.js]: This line seems unnecessary?

Comment (Zarel) [inline team-validator.js]: You can just remove `customRules` from the crash message.

Comment (sirDonovan) [inline team-validator.js]: Right, I removed it.

**Comment by Zarel** [comment]
Looks great!

**Pull request merged**